### ISSUE TITLE
plot order

### ISSUE THREAD

**Issue by wangjiawen2013** [scverse/scanpy#906]
Dear,
sc.pl.umap() and other related functions seem to plot the points in a certain order. For example, If I have two clusters A and B, then sc.pl.umap(adata,color=['leiden'],groups='A') would would plot cluster A first in blue color, leaving points of cluster B as grey color, this is perfect. However, if I wanna show cluster B with sc.pl.umap(adata,color=['leiden'],groups='B'), cluster A will be shown in grey color, and cluster B will be shown in yellow color, but points of cluster A will overlap with cluster B, as a result, some points of B are blocked by A. So, it's very helps if the user can set the display order mannually.

**Comment by gokceneraslan** [comment]
This is a duplicate of https://github.com/theislab/scanpy/issues/891, I think, which is already fixed in master branch.

**Comment by wangjiawen2013** [comment]
Yes, I installed the development version of scanpy, and it worked!

**Comment by wangjiawen2013** [comment]
> This is a duplicate of #891, I think, which is already fixed in master branch.

It is fixed in sc.pl.umap(), but still exist in sc.pl.diffmap(), if not all of the groups are used, for example, sc.pl.diffmap(adata, color=['leiden'],size=20,components=['4,5'],groups=['0','1','2','3']), the rest groups  4, 5, 6 ....... will be gray and cover the points of group 0, 1, 2, 3
